lib/rubocop/cop/rails/delegate_allow_blank.rb

```diff
@@ -0,0 +1,59 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop looks for delegations that pass :allow_blank as an option
+      # instead of :allow_nil. :allow_blank is not a valid option to pass
+      # to ActiveSupport's :delegate method.
+      #
+      # @example
+      #   # bad
+      #   delegate :foo, to: :bar, allow_blank: true
+      #
+      #   # good
+      #   delegate :foo, to: :bar, allow_nil: true
+      class DelegateAllowBlank < Cop
+        MSG = '`allow_blank` is not a valid option, use `allow_nil`'.freeze
+
+        def_node_matcher :delegate, <<-PATTERN
+          (send _ :delegate _ $hash)
+        PATTERN
+
+        def_node_matcher :delegate_options, <<-PATTERN
+          (hash $...)
+        PATTERN
+
+        def_node_matcher :allow_blank?, <<-PATTERN
+          (pair $(sym :allow_blank) true)
+        PATTERN
+
+        def on_send(node)
+          return unless (offending_node = allow_blank_option(node))
```

Comment:
Btw, I'm not a huge fan of assignments in conditions - guess you can move this before the `return`.